pkg/client/fake/client.go

```diff
@@ -56,18 +56,11 @@ const (
 	maxGeneratedNameLength = maxNameLength - randomLength
 )
 
-// NewFakeClient creates a new fake client for testing.
-// You can choose to initialize it with a slice of runtime.Object.
-// Deprecated: use NewFakeClientWithScheme.  You should always be
-// passing an explicit Scheme.
-func NewFakeClient(initObjs ...runtime.Object) client.Client {
-	return NewFakeClientWithScheme(scheme.Scheme, initObjs...)
-}
-
-// NewFakeClientWithScheme creates a new fake client with the given scheme
+// NewFakeClient creates a new fake client with the given scheme
 // for testing.
+//
 // You can choose to initialize it with a slice of runtime.Object.
-func NewFakeClientWithScheme(clientScheme *runtime.Scheme, initObjs ...runtime.Object) client.Client {
+func NewFakeClient(clientScheme *runtime.Scheme, initObjs ...runtime.Object) client.Client {
```

Comment:
Should we remove `initObjs` support as well here? It creates object without a resourceVersion which is technically incorrect

Reply:
I would like to keep it, its super convenient. If anything, we can extend this to set the resourceVersion

Reply:
Yeah, there was another PR out to do that, although that breaks lots of tests